csrc/fusion.cpp

```diff
@@ -116,6 +116,14 @@ IrCloner Fusion::copy(const Fusion* from, Fusion* to) {
 
   to->expected_dynamic_smem_bytes_ = from->expected_dynamic_smem_bytes_;
 
+  if (from->all_tvs_ptr_ != nullptr) {
```

Comment:
nit: avoid unnecessary copy of a stale vector
```suggestion
  if (from->all_tvs_and_uses_valid_) {
```